models/aeif_psc_delta.cpp

```diff
@@ -0,0 +1,533 @@
+
+  B_.I_stim_ = 0.0;
+}
+
+void
+nest::aeif_psc_delta::calibrate()
+{
+  // ensures initialization in case mm connected after Simulate
+  B_.logger_.init();
+
+  // set the right threshold and GSL function depending on Delta_T
+  if ( P_.Delta_T > 0. )
+  {
+    V_.V_peak_ = P_.V_peak_;
+  }
+  else
+  {
+    V_.V_peak_ = P_.V_th; // same as IAF dynamics for spikes if Delta_T == 0.
+  }
+
+  V_.refractory_counts_ = Time( Time::ms( P_.t_ref_ ) ).get_steps();
+  // since t_ref_ >= 0, this can only fail in error
+  assert( V_.refractory_counts_ >= 0 );
+}
+
+/* ----------------------------------------------------------------
+ * Update and spike handling functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::aeif_psc_delta::update( const Time& origin,
+  const long from,
+  const long to )
+{
+  assert(
+    to >= 0 && ( delay ) from < kernel().connection_manager.get_min_delay() );
+  assert( from < to );
+  assert( State_::V_M == 0 );
+  const double h = Time::get_resolution().get_ms();
+  const double tau_m_ = P_.C_m / P_.g_L;
+  for ( long lag = from; lag < to; ++lag )
+  {
+    double t = 0.0;
+
+    if ( S_.r_ > 0 )
+      --S_.r_;
+
+    // numerical integration with adaptive step size control:
+    // ------------------------------------------------------
+    // gsl_odeiv_evolve_apply performs only a single numerical
+    // integration step, starting from t and bounded by step;
+    // the while-loop ensures integration over the whole simulation
+    // step (0, step] if more than one integration step is needed due
+    // to a small integration step size;
+    // note that (t+IntegrationStep > step) leads to integration over
+    // (t, step] and afterwards setting t to step, but it does not
+    // enforce setting IntegrationStep to step-t
+    while ( t < B_.step_ )
+    {
+      const int status = gsl_odeiv_evolve_apply( B_.e_,
+        B_.c_,
+        B_.s_,
+        &B_.sys_,             // system of ODE
+        &t,                   // from t
+        B_.step_,             // to t <= step
+        &B_.IntegrationStep_, // integration step size
+        S_.y_ );              // neuronal state
+
+      if ( status != GSL_SUCCESS )
+      {
+        throw GSLSolverFailure( get_name(), status );
+      }
+      // check for unreasonable values; we allow V_M to explode
+      if ( S_.y_[ State_::V_M ] < -1e3 || S_.y_[ State_::W ] < -1e6
+        || S_.y_[ State_::W ] > 1e6 )
+      {
+        throw NumericalInstability( get_name() );
+      }
+      // spikes are handled inside the while-loop
+      // due to spike-driven adaptation
+      if ( S_.r_ == 0 )
+      {
+        // neuron not refractory
+        S_.y_[ State_::V_M ] =
+          S_.y_[ State_::V_M ] + B_.spikes_.get_value( lag );
+
+        // if we have accumulated spikes from refractory period,
+        // add and reset accumulator
+        if ( P_.with_refr_input_ && S_.refr_spikes_buffer_ != 0.0 )
```

Comment:
@Silmathoron You are quite right in principle, but some user require this and therefore we decided after a lengthy discussion some time ago to support this as a non-default option. The argument mainly was that input into neurons with alpha or exponential currents during the refractory period is not lost entirely, as it starts input currents. For compatibility with `iaf_psc_delta`, `aeif_psc_delta` therefore should support this, too.